### PR TITLE
[1LP][RFR] Refactored orchestration templates, fixed azure issues

### PR DESCRIPTION
**Pull request by jteehan**
Purpose or Intent
=================

__Fixing__ 
https://projects.engineering.redhat.com/browse/RHCFQE-1693
Cleanup was busted.  Someone was trying to run a method on a string

https://projects.engineering.redhat.com/browse/RHCFQE-1712
Azure dying at template create

and

__Extending__ 
https://projects.engineering.redhat.com/browse/RHCFQE-1750
Extracted data to separate files as request in PR

Fixed all the missing Azure stack data entries
Fixed all the missing form field entries.

{{pytest: cfme/tests/services/test_provision_stack.py::test_provision_stack -v --use-provider azure }}

### CODE REVIEW

**Comment by jteehan** [comment]
I don't think this will pass PRT without https://gitlab.cee.redhat.com/cfme-qe/cfme-qe-yamls/merge_requests/82

File: cfme/services/catalogs/orchestration_template.py [hunk 1]
```diff
@@ -140,8 +140,10 @@ def create(self, content):
         view = navigate_to(self, "AddTemplate")
         if(self.template_type == "CloudFormation Templates"):
             temp_type = "Amazon CloudFormation"
-        else:
+        elif(self.template_type == "Heat Templates"):
```
Comment (RonnyPfannschmidt) [inline cfme/services/catalogs/orchestration_template.py]: there no longer is a default case, should there be a catch all or an error?

if there is a chance this mapping will grow, we should consider putting the type mapping into an actual dictionary

**Comment by RonnyPfannschmidt** [comment]
overall this one looks nice, good work 👍

**Comment by jteehan** [comment]
PRT passed across the board as expected.  This is my first pass at something new so I would expect at least a few issues.  Thanks in advance.

File: cfme/services/catalogs/orchestration_template.py (hunk 1 above)
Comment (jteehan) [inline cfme/services/catalogs/orchestration_template.py]: Good point.  There is no default other than to raise an exception I suppose.  That would probably be best.  This list has been pretty static for a long time.

Comment (jteehan) [inline cfme/services/catalogs/orchestration_template.py]: maybe
        else:
            raise Exception("ERROR template_type needs to be one of the above")

File: cfme/tests/services/test_provision_stack.py [hunk 2]
```diff
@@ -197,21 +36,18 @@
...
-    if provider.type == 'azure':
-        template_name = 'azure-single-vm-from-user-image'
-    else:
-        template_name = fauxfactory.gen_alphanumeric()
-
+    template_name = fauxfactory.gen_alphanumeric()
     template = OrchestrationTemplate(template_type=template_type,
                                      template_name=template_name)
 
     if provider.type == "ec2":
```
Comment (mshriver) [inline cfme/tests/services/test_provision_stack.py]: If all the data file names match the provider type strings, you can condense this into a single call that parses in the provider type `'{}_vm_template.data'.format(provider.type)`

File: cfme/tests/services/test_provision_stack.py [hunk 3]
```diff
@@ -337,7 +177,6 @@ def _cleanup_vms():
             if provider.mgmt.stack_exist(stack_data['stack_name']):
                 wait_for(lambda: provider.mgmt.delete_stack(stack_data['stack_name']),
                  delay=10, num_sec=800, message="wait for stack delete")
-            stack_data['vm_name'].delete_from_provider()
```
Comment (mshriver) [inline cfme/tests/services/test_provision_stack.py]: Looks like just removing this is fine as the if block above should take care of the cleanup. :+1:

File: cfme/services/catalogs/orchestration_template.py [hunk 4]
```diff
@@ -140,8 +140,12 @@ def create(self, content):
...
             temp_type = "OpenStack Heat"
+        elif(self.template_type == "Azure Templates"):
```
Comment (izapolsk) [inline cfme/services/catalogs/orchestration_template.py]: @jteehan, braces aren't needed in Python. this isn't C

Comment (mshriver) [inline cfme/services/catalogs/orchestration_template.py]: I do this too occasionally but only leave it if I've got multiple statements grouped.

File: cfme/tests/services/test_provision_stack.py (hunk 3 above)
Comment (jteehan) [inline cfme/tests/services/test_provision_stack.py]: stack_data['vm_name'] is a string, so I don't know what everything after the . does except throw a warning in the log. But yes, it does get cleaned up fine.

File: cfme/tests/services/test_provision_stack.py (hunk 2 above)
Comment (jteehan) [inline cfme/tests/services/test_provision_stack.py]: Azure is a special template which really can't be created on the fly.  Plus, in the future we're going to want more templates for error checking scenarios as well.  And that's just for VMs.  There are all kinds of orchestration templates that aren't being created.  Networks, storage, security groups.  I see what you're saying, but that's why I added _vm_ to the template name in anticipation of future scaling.

File: cfme/services/catalogs/orchestration_template.py (hunk 4 above)
Comment (jteehan) [inline cfme/services/catalogs/orchestration_template.py]: I didn't put them there  :)  Changed.

**Comment by dajoRH** [comment]
Lint report for commit 3be7026102b1045e98ca8d52cc0c3242f4e072f1:

`cfme/services/catalogs/orchestration_template.py`:
- :bangbang: Line 141:60: **E203** *whitespace before ':'*
- :bangbang: Line 143:52: **E203** *whitespace before ':'*
- :bangbang: Line 145:53: **E203** *whitespace before ':'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by jteehan** [comment]
I think I answered or changed everything in the list.  Love to get it merged before my 1:1 tomorrow.

**Comment by mshriver** [comment]
@izapolsk only had 1 optional comment from what I see, so I tagged [1LP] after my re-review

**Pull request merged**